Add Vale rule reference page (#755)
grafana/writers-toolkit: docs/sources/review/lint-prose/rules.md

---
date: "2024-06-25"
description: A description of every Grafana Labs prose linting rule.
menuTitle: Rules
review_date: "2024-06-25"
title: Vale rules
---

# Vale rules

<!-- These are our style rules. -->
<!-- vale Grafana.We = NO -->

Vale codifies our style guide into a series of rules that can be checked against your prose.

> The following is a list of all the rules that we have defined.

Reviewer: Since you also like being a nitpicker except in real time while I'm trying to write a document. Troll.
```suggestion
The following is a list of all the rules that we've defined.
```

Extends: substitution

When linking in Markdown, use '%s' instead of '%s'.

[More information ->](https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#links-and-references)

### Grafana.RepeatedWords

Extends: repetition

'%s' is repeated

### Grafana.Spelling

Extends: spelling

Did you really mean '%s'?

The Grafana dictionary might not know of this word yet.

To add a new word, refer to [Add words to the Grafana Labs dictionary](https://grafana.com/docs/writers-toolkit/review/lint-prose/dictionary/add-words/).
Alternatively, raise an [issue](https://github.com/grafana/writers-toolkit/issues/new?title=Grafana.Spelling%%3A%%20%[1]s) and a maintainer will add it for you.

For UI elements, use [bold formatting](https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#bold).
The spell checker doesn't check words with bold formatting.

For paths; configuration; user input; code; class, method, and variable names; statuscodes; and console output, use [code formatting](https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#bold).
The spell checker doesn't check words with code formatting.

## Warnings

> 

Reviewer 2: add intro "The following sections define ..."

### Grafana.Admin

Extends: substitution

Use administrator instead of admin unless it's the name of the UI label like in the Grafana 'Admin' role.

[More information ->](https://developers.google.com/style/word-list#admin)

### Grafana.Admonitions

Extends: script

Prefer the `admonition` shortcode over blockquotes.

The admonition shortcode renders its content in a blockquote with consistent styling across the website.

[More information ->](https://grafana.com/docs/writers-toolkit/write/shortcodes/#admonition)

### Grafana.Agentless

Extends: substitution

Grafana Agent has been replaced by Grafana Alloy, so you shouldn't use agent-based terminology.

> If you're talking about why and how to send signals directly from app to prefer no-collector to agentless.

Reviewer: "from app to"? I'm not sure what the to is doing. Also added a needed comma.
```suggestion
If you're talking about why and how to send signals directly from app to, prefer no-collector to agentless.
```

This is consistent with [OTel documentation](https://opentelemetry.io/docs/collector/deployment/no-collector/).

[More information ->](https://grafana.com/docs/writers-toolkit/write/style-guide/word-list/#no-collector)

### Grafana.AllowsTo

Extends: substitution

Did you mean '%s' instead of '%s'?

Allows to is a common wording error.

### Grafana.AltText

Extends: script

All images must have alt text.

[More information ->](https://grafana.com/docs/writers-toolkit/write/image-guidelines/#alt-text)

### Grafana.AmazonCloudWatch

Extends: substitution

Use '%s' instead of '%s'.

[More information ->](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/WhatIsCloudWatch.html)

### Grafana.AmazonProductNames

Extends: conditional

Use the full Amazon product name in the first instance.

[More information ->](https://grafana.com/docs/writers-toolkit/write/style-guide/capitalization-punctuation/#amazon-products)

### Grafana.AndOr

Extends: existence

Avoid writing and/or except when space is limited, such as in tables.

> Often, and implies or, so you don't need to write both words.

Reviewer: ```suggestion
Often, 'and' implies 'or', so you don't need to write both words.
```

If you need to specify both in your content, write something like "You can export raw events, processed events, or both."

[More information ->](https://developers.google.com/style/slashes#and-or)

### Grafana.CHANGELOG

Extends: substitution

> Use '%s' instead of '%s' unless you are referring to a specific file which has that spelling.

Reviewer: ```suggestion
Use '%s' instead of '%s' unless you're referring to a specific file which has that spelling.
```

[More information ->](https://grafana.com/docs/writers-toolkit/write/style-guide/word-list/#changelog)

### Grafana.CommandLinePrompts

Extends: script

Don't add `$` or `#` as prompts before commands. Make it easy for users to copy and paste commands.

> Also don't use `#` to include comments in commands. That explanation should be outside of the code block.

Reviewer: ```suggestion
Also, don't use `#` to include comments in commands. That explanation should be outside of the code block.
```

### Grafana.Kubernetes

Extends: substitution

Use '%s' instead of '%s'.

[More information ->](https://grafana.com/docs/writers-toolkit/write/style-guide/capitalization-punctuation/#kubernetes-objects)

### Grafana.ProductPossessives

Extends: existence

Don't form a possessive from a feature name, product name, or trademark, regardless of who owns it.
Instead, use the name as a modifier or rewrite to use a word like of to indicate the relationship.

[More information ->](https://developers.google.com/style/possessives#product,-feature,-and-company-names)

### Grafana.PrometheusExporters

Extends: substitution

Use '%s' instead of '%s'.

[More information ->](https://grafana.com/docs/writers-toolkit/write/style-guide/word-list/#node-exporter)

### Grafana.Quickstart

Extends: substitution

Use the compound adjective without a hyphen whether the noun is implied or explicit. For example, you can use _quickstart guide_ or just _quickstart_.

[More information ->](https://grafana.com/docs/writers-toolkit/write/style-guide/word-list/#quickstart)

### Grafana.README

Extends: substitution

> Use '%s' instead of '%s' unless you are referring to a specific file which has that spelling.

Reviewer: ```suggestion
Use '%s' instead of '%s' unless you're referring to a specific file which has that spelling.
```

[More information ->](https://grafana.com/docs/writers-toolkit/write/style-guide/word-list/#readme)

### Grafana.React

Extends: substitution

Use '%s' instead of '%s'.

[More information ->](https://grafana.com/docs/writers-toolkit/write/style-guide/word-list/#react)

### Grafana.SQL

Extends: substitution

Use '%s' instead of '%s'.

The article—a or an—that you use before the acronym SQL depends on how the word is pronounced.

> When referring to the product Microsoft SQL Server, SQL should be pronounced "sequel". In this case, use the article a, as in a SQL Server analysis.

Reviewer: ```suggestion
When referring to the product Microsoft SQL Server, SQL should be pronounced "sequel". In this case, use the article 'a', as in a SQL Server analysis.
```

> When referring to the term in any other context, such as SQL databases, errors, or servers, SQL should be pronounced "ess-cue-el". In this case, use the article an, as in an SQL error.

Reviewer: ```suggestion
When referring to the term in any other context, such as SQL databases, errors, or servers, SQL should be pronounced "ess-cue-el". In this case, use the article 'an', as in an SQL error.
```

[More information ->](https://grafana.com/docs/writers-toolkit/write/style-guide/word-list/#sql-structured-query-language)

### Grafana.Shortcodes

Extends: script

Prefer `{{</* admonition type="<TYPE>" */>}}`.

It has the most consistent semantics.

The percent syntax is used for special behavior that isn't required with this shortcode.

[More information ->](https://grafana.com/docs/writers-toolkit/write/shortcodes/#admonition)

### Grafana.We

Extends: existence

Use first person plural pronouns like '%s' carefully.

Don't use 'we' when you're talking about the reader, instead use 'you'.

It's OK to use 'we' when you're talking about Grafana Labs.

[More information ->](https://developers.google.com/style/person#use-first-person-plural-pronouns-carefully)

### Grafana.Wish

Extends: substitution

Use '%s' instead of '%s'.

[More information ->](https://developers.google.com/style/word-list#wish)

### Grafana.WordList

Extends: substitution

Use '%s' instead of '%s'.

[More information ->](https://developers.google.com/style/word-list)

## Suggestions

### Grafana.Acronyms

Extends: conditional

> Spell out '%s', if it is unfamiliar to the audience.

Reviewer: ```suggestion
Spell out '%s', if it's unfamiliar to the audience.
```
